### PR TITLE
Merge `8.2.x` into `develop`

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Merge `8.2.x` into `develop`
| Type?             | improvement
| Category?         | ME
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI + Tests UI
| UI Tests          | https://github.com/boherm/ga.tests.ui.pr/actions/runs/11103487385
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Comment by ps-jarvis** [comment]
This pull request seems to contain new translation strings. I have summarized them below to ease up review:

- `Admin.Modules.Notification`
    - [ ] `The override file %1$s conflicts with an existing override in %2$s.`

(Note: this is an automated message, but answering it will reach a real human)

**Review by nicosomb — APPROVED** [review]

**Review by nicosomb — DISMISSED** [review]
CI is 🔴

File: controllers/admin/AdminCartsController.php
Comment (jolelievre) [inline controllers/admin/AdminCartsController.php:1]: This controller has been migrated so this file should not be integrated back in the develop branch and must be deleted

File: src/PrestaShopBundle/Resources/config/services/core/module.yml [hunk 1]
```diff
@@ -55,6 +55,19 @@ services:
   prestashop.module.factory.sourcehandler:
     alias: PrestaShop\PrestaShop\Core\Module\SourceHandler\SourceHandlerFactory
     public: true
-    deprecated:
-      package: PrestaShop\PrestaShop
-      version: 8.1
+    deprecated: ~
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/services/core/module.yml]: We should keep the 8.1 indication to know when this was deprecated, so we can remove it in the next version
Actually, this should be removed in the v9.0 but in a dedicated PR probably

File: translations/default/ModulesBlockreassuranceAdmin.xlf [hunk 2]
```diff
@@ -151,15 +151,6 @@ Comment: Confirm uninstall</note>
       </trans-unit>
     </body>
   </file>
-  <file original="modules/blockreassurance/views/templates/admin/tabs/content/config_elements/cms.tpl" source-language="en-US" target-language="en" datatype="plaintext">
```
Comment (jolelievre) [inline translations/default/ModulesBlockreassuranceAdmin.xlf]: You should modify absolutely NO translation files, you can revert the changes on all those files

**Review by jolelievre — CHANGES_REQUESTED** [review]

File: classes/Customer.php [hunk 3]
```diff
@@ -1221,17 +1221,20 @@ public function transformToCustomer($idLang, $password = null)
...
-            $this->passwd = $crypto->hash(Tools::passwdGen(16, 'RANDOM'));
+            $password = Tools::passwdGen(16, 'RANDOM');
         } else {
             if (!Validate::isAcceptablePasswordLength($password) || !Validate::isAcceptablePasswordScore($password)) {
                 return false;
             }
-            $this->passwd = $crypto->hash($password);
         }
 
+        /** @var \PrestaShop\PrestaShop\Core\Crypto\Hashing $crypto */
```
Comment (Hlavtox) [inline classes/Customer.php]: Please don't merge anything in this method, it's already fixed on develop branch.

**Review by Hlavtox — CHANGES_REQUESTED** [review]
See comment, please do not change Customer.php.

**Review by Hlavtox — DISMISSED** [review]
All good now. :-) thanks @boherm

**Review by matthieu-rolland — DISMISSED** [review]
thanks @boherm 👍

File: src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php [hunk 4]
```diff
@@ -92,24 +92,7 @@ class CmsPageController extends FrameworkBundleAdminController
...
-                'error',
-                $this->getErrorMessageForException($exception, $this->getErrorMessages())
-            );
-
-            if ($exception instanceof CmsPageCategoryNotFoundException) {
-                return $this->redirectToRoute('admin_cms_pages_index');
-            }
-        }
-
+        $viewData = $this->get('prestashop.core.cms_page.data_provider.cms_page_view')->getView($cmsCategoryParentId);
```
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php:95]: Lucky for you it was not refactored yet ^^

File: tests/Integration/Core/Addon/Module/ModuleManagerBuilderTest.php [hunk 5]
```diff
@@ -129,29 +163,22 @@ public function testInstall(): void
...
-        $expected_override_admin_product = $this->cleanup($expected_override_admin_product);
+        $actual_override_admin_product = $this->cleanup(file_get_contents(_PS_ROOT_DIR_ . '/override/controllers/admin/DummyAdminController.php'));
+        $expected_override_admin_product = $this->cleanup(file_get_contents($resource_path . '/controllers/admin/DummyAdminController.php'));
 
         $this->assertEquals(
-            $expected_override_admin_product,
             $actual_override_admin_product,
-            'DummyAdminController.php file different'
+            $expected_override_admin_product,
+            'AdminProductsController.php file different'
```
Comment (jolelievre) [inline tests/Integration/Core/Addon/Module/ModuleManagerBuilderTest.php]:
```suggestion
            'DummyAdminController.php file different'
```

**Review by jolelievre — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**